agent/stats/service_connect_linux.go

```diff
@@ -66,26 +72,163 @@ func (sc *ServiceConnectStats) retrieveServiceConnectStats(task *apitask.Task) {
+					if math.IsInf(*bucket.UpperBound, 0) {
+						continue
+					}
+					metricValues = append(metricValues, bucket.UpperBound)
+					metricCount := int64(*bucket.CumulativeCount)
+					metricCounts = append(metricCounts, &metricCount)
+
+				}
+				metricCounts = convertHistogramMetricCounts(metricCounts)
+			} else {
+				seelog.Errorf("service connect stats received invalid Metric type: %v", v.Type.String())
+			}
+
+			// Get metric dimensions
+			if metric.Label != nil {
+				for _, d := range metric.Label {
+					// TODO [SC]: New stats endpoint will surface a new dimension which indicates what is the MetricType.
+					dimension := &ecstcs.Dimension{
+						Key:   d.Name,
+						Value: d.Value,
+					}
+					dimensions = append(dimensions, dimension)
+				}
+			}
+
+			generalMetric.MetricName = v.Name
+			generalMetric.MetricValues = metricValues
+			generalMetric.MetricCounts = metricCounts
+
+			// Check if dimensions exist in statsCollectedList already
+			generalMetricsWrapper := checkIfDimensionExist(dimensions, statsCollectedList)
+			// Dimension does not exist in statsCollectedList, add it to the statsCollectedList
+			if generalMetricsWrapper == nil {
+				generalMetricsList := []*ecstcs.GeneralMetric{generalMetric}
+				generalMetricsWrapper = &ecstcs.GeneralMetricsWrapper{
+					Dimensions:     dimensions,
+					GeneralMetrics: generalMetricsList,
+				}
+				statsCollectedList = append(statsCollectedList, generalMetricsWrapper)
+			} else {
+				// Add this metric to the metric list for the already existing dimesion
+				generalMetricsWrapper.GeneralMetrics = append(generalMetricsWrapper.GeneralMetrics, generalMetric)
+			}
+		}
+	}
+
+	sc.resetStats()
+	sc.setStats(statsCollectedList)
+	sc.SetStatsSent(false)
 }
 
-func (sc *ServiceConnectStats) setStats(stats string) {
+func (sc *ServiceConnectStats) setStats(stats []*ecstcs.GeneralMetricsWrapper) {
 	sc.lock.Lock()
 	defer sc.lock.Unlock()
 
 	sc.stats = stats
 }
 
-func (sc *ServiceConnectStats) GetStats() string {
+func (sc *ServiceConnectStats) GetStats() []*ecstcs.GeneralMetricsWrapper {
 	sc.lock.RLock()
 	defer sc.lock.RUnlock()
 
 	return sc.stats
 }
+
+func (sc *ServiceConnectStats) SetStatsSent(sent bool) {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+
+	sc.sent = sent
+}
+
+func (sc *ServiceConnectStats) HasStatsBeenSent() bool {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+
+	return sc.sent
+}
+
+func (sc *ServiceConnectStats) resetStats() {
+	sc.lock.RLock()
+	defer sc.lock.RUnlock()
+
+	sc.stats = nil
+}
+
+// CloudWatch accepts the histogram buckets in a disjoint manner while the prometheus emits these values in a cumulative way.
+// This method performs that conversion
+func convertHistogramMetricCounts(metricCounts []*int64) []*int64 {
+	prevCount := metricCounts[0]
+
+	for i := 1; i < len(metricCounts); i++ {
+		newCount := *metricCounts[i] - *prevCount
+		prevCount = metricCounts[i]
+		metricCounts[i] = &newCount
+	}
```

Comment:
nit: this can be expressed in a more idiomatic way as follows (note `prevCount` is initialized differently): 

```golang
prevCount := *metricCounts[0]
for i := 1; i < len(metricCounts); i++ {
    prevCount, *metricCounts[i] = *metricCounts[i], *metricCounts[i] - prevCount
}
```